apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -42,14 +55,93 @@ func (host *BareMetalHost) validateChanges(old *BareMetalHost) []error {
 	return errs
 }
 
-func validateRAID(r *RAIDConfig) error {
-	if r == nil {
+func validateRAID(s BareMetalHostSpec) error {
+	if s.RAID == nil {
 		return nil
 	}
 
-	if len(r.HardwareRAIDVolumes) > 0 && len(r.SoftwareRAIDVolumes) > 0 {
+	if len(s.RAID.HardwareRAIDVolumes) > 0 && len(s.RAID.SoftwareRAIDVolumes) > 0 {
 		return fmt.Errorf("hardwareRAIDVolumes and softwareRAIDVolumes can not be set at the same time")
 	}
 
+	bmcType, err := getBMCType(s.BMC.Address)
+	if err != nil {
+		return err
+	}
+
+	switch bmcType {
+	case "ilo5", "idrac", "irmc":
+		return nil
+	default:
+		return fmt.Errorf("currently the 'raid' field is only supported by ilo5/idrac/irmc")
+	}
+}
+
+func getBMCType(address string) (string, error) {
```

Comment:
It looks like we should use the same parse logic https://github.com/metal3-io/baremetal-operator/blob/master/pkg/ironic/bmc/access.go#L74 which does more stuff and less error prone.